Alex_RTOS_Sailboat/RTOS_framework_v1/src/sail_nmea.h

```diff
@@ -11,18 +11,222 @@
 
 #include <status_codes.h>
 #include <stdint.h>
+#include <stdbool.h>
 
 #include "sail_uart.h"
 
 #define NMEA_BUFFER_LENGTH 128
+#define PREFIX_LIM 6
 
 typedef enum NMEA_ChannelIDs {
 	NMEA_GPS,
-	NMEA_WEATHERSTATION,
+	//NMEA_WEATHERSTATION,
 	NMEA_RADIO,
 	NMEA_NUM_CHANNELS
 } NMEA_ChannelID;
 
+/********** NMEA Message Types **********/
+
+/* List the various NMEA message types */
+typedef enum eNMEA_TRX {
+	eGPGGA,
+	eWIMWV,
+	eYXXDR,
+	eHCHDT,
+	//the rest of the types are listed below, but not currently used
+	/*
+	eGPVTG,
+	eGPDTM,		
+	eGPGLL,		
+	eGPGSA,   
+	eGPGSV,		
+	eGPRMC,		
+	eGPVTG,    
+	eGPZDA,
+	eHCHDT,		
+	eHCHDG,			
+	eHCTHS,		
+	eTIROT,		
+	eWIMDA,		
+	eWIMWV,
+	eWIMWR,
+	eWIMWT,	
+	*/
+	NUM_NMEA_TYPES
+} eNMEA_TRX_t;
+
+typedef struct NMEA_TYPES_INFO {
+	eNMEA_TRX_t MSG_id;
+	char NMEA_Prefix[PREFIX_LIM];
+} NMEA_TYPE_MAP;
+
+static NMEA_TYPE_MAP NMEA_TYPE_TABLE[NUM_NMEA_TYPES] = {
+	{ eGPGGA, "GPGGA"}, 
+	{ eWIMWV, "WIMWV"}, 
+	{ eYXXDR, "YXXDR"}, 
+	{ eHCHDT, "HCHDT"}
+	
+	//remaining fields
+	/*
+	{ eGPDTM, "GPDTM"}, 
+	{ eGPGLL, "GPGLL"}, 
+	{ eGPGSA, "GPGSA"},
+	{ eGPGSV, "GPGSV"}, 
+	{ eGPRMC, "GPRMC"},
+	{ eGPVTG, "GPVTG"}, 
+	{ eGPZDA, "GPZDA"},
+	{ eHCHDG, "HCHDG"}, 
+	{ eHCTHS, "HCTHS"}, 
+	{ eTIROT, "TIROT"},
+	{ eWIMDA, "WIMDA"}, 
+	{ eWIMWV, "WIMWV"}, 
+	{ eWIMWR, "WIMWR"},
+	{ eWIMWT, "WIMWT"}, 
+	{ eYXXDR, "YXXDR"},
+		*/
+};
+
+enum west_east {west, east};
+enum north_south {north, south};	
+	
+typedef struct latitude_info {
+	float lat;
+	enum north_south ns;
+} latitude;
+
+typedef struct longitude_info {
+	float lon;
+	enum west_east we;
+} longitude;
```

Comment:
Should the lat/long stuff be in gps or here?

Reply:
Intuitively it makes more sense to have these in the gps, but they are in the NMEA because they are specific to the types NMEA messages. See lines 102-106. If we put the longitude_info in the gps file, the NMEA files would depend on the gps files (while the gps files also depend on the NMEA files). Not sure if that's a good idea (increases coupling)